lib/credentials/shared_ini_file_credentials.js

```diff
@@ -48,11 +51,11 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
 
     this.filename = options.filename;
     this.profile = options.profile || process.env.AWS_PROFILE || 'default';
-    this.get(function() {});
```

Comment:
I made the change, but the problem here is that the standard provider chain will result in multiple role assumptions. The reason being that the constructor initiates a role assumption and then any api calls will start another when they get credentials (since the initial one has not completed). Is there any way to address that issue? Another constructor argument to disable the initialization `get`?